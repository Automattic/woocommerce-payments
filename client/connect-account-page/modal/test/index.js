/**
 * External dependencies
 */
import { render } from '@testing-library/react';

/**
 * Internal dependencies
 */
import OnboardingLocationCheckModal from '../index';

// Mocks, needed for invoking the modal rendering
const countriesMock = [
	{ title: 'United Kingdom (UK)' },
	{ title: 'United States (US)' },
];
const callbackMock = () => {};

// Utility function for accessing the modal content in assertions
const modalContent = () => {
	return document.body.getElementsByClassName(
		'woocommerce-payments__onboarding_location_check-modal'
	);
};

describe( 'Onboarding: location check dialog', () => {
	beforeEach( () => {
		jest.clearAllMocks();
	} );

	test( 'renders correctly when opened', () => {
		render(
			<OnboardingLocationCheckModal
				countries={ countriesMock }
				whenConfirmed={ callbackMock }
				whenDeclined={ callbackMock }
			/>
		);
		expect( modalContent() ).toMatchSnapshot();
	} );
} );
