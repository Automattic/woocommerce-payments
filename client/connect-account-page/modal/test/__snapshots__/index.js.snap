// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Onboarding: location check dialog renders correctly when opened 1`] = `
HTMLCollection [
  <div
    aria-labelledby="components-modal-header-0"
    class="components-modal__frame woocommerce-payments__onboarding_location_check-modal"
    role="dialog"
    tabindex="-1"
  >
    <div
      class="components-modal__content"
      role="document"
    >
      <div
        class="components-modal__header"
      >
        <div
          class="components-modal__header-heading-container"
        >
          <h1
            class="components-modal__header-heading"
            id="components-modal-header-0"
          >
            WooCommerce Payments
          </h1>
        </div>
        <button
          aria-label="Close dialog"
          class="components-button has-icon"
          type="button"
        >
          <svg
            aria-hidden="true"
            focusable="false"
            height="24"
            role="img"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M13 11.8l6.1-6.3-1-1-6.1 6.2-6.1-6.2-1 1 6.1 6.3-6.5 6.7 1 1 6.5-6.6 6.5 6.6 1-1z"
            />
          </svg>
        </button>
      </div>
      <div
        class="woocommerce-payments__onboarding_location_check-wrapper"
      >
        <div
          class="woocommerce-payments__onboarding_location_check-modal-message"
        >
          It appears you're attempting to set up WooCommerce Payments from an unsupported country. In order to complete the set up of WooCommerce Payments, your store is required to have a business entity in one of the following countries: 
          <ul
            class="woocommerce-list"
            role="menu"
          >
            <li
              class="woocommerce-list__item"
            >
              <div
                class="woocommerce-list__item-inner"
              >
                <div
                  class="woocommerce-list__item-text"
                >
                  <span
                    class="woocommerce-list__item-title"
                  >
                    United Kingdom (UK)
                  </span>
                </div>
              </div>
            </li>
            <li
              class="woocommerce-list__item"
            >
              <div
                class="woocommerce-list__item-inner"
              >
                <div
                  class="woocommerce-list__item-text"
                >
                  <span
                    class="woocommerce-list__item-title"
                  >
                    United States (US)
                  </span>
                </div>
              </div>
            </li>
          </ul>
           
          <a
            data-link-type="external"
            href="https://docs.woocommerce.com/document/payments/countries/"
            rel="noopener noreferrer"
            target="_blank"
          >
            Learn more
          </a>
           about setting up business entities in foreign countries.
        </div>
        <div
          class="woocommerce-payments__onboarding_location_check-footer"
        >
          <button
            class="components-button is-secondary"
            type="button"
          >
            Continue
          </button>
          <button
            class="components-button is-primary"
            type="button"
          >
            Cancel
          </button>
        </div>
      </div>
    </div>
  </div>,
]
`;
